Modules/StudyProgramme/test/model/Progress/ilStudyProgrammeProgressTest.php

```diff
@@ -202,4 +209,241 @@ public function test_invalidate_non_expired_2()
+        $spp = (new ilStudyProgrammeProgress(123))
+            ->markAccredited($now, $usr);
+
+        $this->assertEquals($now, $spp->getCompletionDate());
+        $this->assertEquals($usr, $spp->getCompletionBy());
+        $this->assertEquals($usr, $spp->getLastChangeBy());
+        $this->assertEquals(
+            $now->format(ilStudyProgrammeProgress::DATE_FORMAT),
+            $spp->getLastChange()->format(ilStudyProgrammeProgress::DATE_FORMAT)
+        );
+        $this->assertEquals(
+            ilStudyProgrammeProgress::STATUS_ACCREDITED,
+            $spp->getStatus()
+        );
+    }
+
+    public function testUnmarkAccredited()
+    {
+        $usr = 6;
+        $now = new DateTimeImmutable();
+        $spp = (new ilStudyProgrammeProgress(123))
+            ->unmarkAccredited($now, $usr);
+
+        $this->assertNull($spp->getCompletionDate());
+        $this->assertNull($spp->getCompletionBy());
+        $this->assertNull($spp->getValidityOfQualification());
+        $this->assertEquals($usr, $spp->getLastChangeBy());
+        $this->assertEquals(
+            $now->format(ilStudyProgrammeProgress::DATE_FORMAT),
+            $spp->getLastChange()->format(ilStudyProgrammeProgress::DATE_FORMAT)
+        );
+        $this->assertEquals(
+            ilStudyProgrammeProgress::STATUS_IN_PROGRESS,
+            $spp->getStatus()
+        );
+    }
+
+
+    public function testSucceed()
+    {
+        $triggering_obj = 777;
+        $now = new DateTimeImmutable();
+        $spp = (new ilStudyProgrammeProgress(123))
+            ->succeed($now, $triggering_obj);
+
+        $this->assertEquals($now, $spp->getCompletionDate());
+        $this->assertEquals($triggering_obj, $spp->getCompletionBy());
+        $this->assertEquals($triggering_obj, $spp->getLastChangeBy());
+        $this->assertEquals(
+            $now->format(ilStudyProgrammeProgress::DATE_FORMAT),
+            $spp->getLastChange()->format(ilStudyProgrammeProgress::DATE_FORMAT)
+        );
+        $this->assertEquals(
+            ilStudyProgrammeProgress::STATUS_COMPLETED,
+            $spp->getStatus()
+        );
+    }
+
+    /**
+     * @dataProvider status
+     */
+    public function testAllowedTransitionsForInProgress($status, $irrelevant)
+    {
+        $spp = (new ilStudyProgrammeProgress(123))->withStatus(ilStudyProgrammeProgress::STATUS_IN_PROGRESS);
+        if (in_array($status, [
+            ilStudyProgrammeProgress::STATUS_NOT_RELEVANT,
+            ilStudyProgrammeProgress::STATUS_ACCREDITED,
+            ilStudyProgrammeProgress::STATUS_FAILED,
+            ilStudyProgrammeProgress::STATUS_COMPLETED
+        ])) {
+            $this->assertTrue($spp->isTransitionAllowedTo($status));
+        } else {
+            $this->assertFalse($spp->isTransitionAllowedTo($status));
+        }
+    }
+
+    /**
+     * @dataProvider status
+     */
+    public function testAllowedTransitionsForAccredited($status, $irrelevant)
```

Comment:
```suggestion
    public function testAllowedTransitionsForAccredited($status, $_)
```

Hab erst gelesen, dass darin ein Flag enthalten sei, ob der Status irrelevant ist.